src/util/pointedthing.cpp

```diff
@@ -59,16 +62,12 @@ void PointedThing::serialize(std::ostream &os) const
 {
 	writeU8(os, 0); // version
 	writeU8(os, (u8)type);
-	if(type == POINTEDTHING_NOTHING)
```

Comment:
in fact remove this is a problem because the else behaviour in incorrect.

Sorry for the previous comment about this.
I think it's better design to use switch case with empty case for NOTHING, this permit to have better code handling. Can you move the two code parts to this mode with NOTHING+empty body ?